### PR TITLE
Channel 1 support

### PR DESCRIPTION
**Pull request by siddharthcb**
I have documented the errors and solutions for the issues encountered for using the channel 1 GPIOs. Hope this is helpful.

### CODE REVIEW

File: ws281x/examples/example.py [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+    
+def colorWipe(red, green, blue, wait_ms=30):
+    for i in range(led_count):
+        set_leds(leds=[LEDState(i, red, green, blue)])
+        time.sleep(wait_ms / 1000.0)
+        
+def wheel(pos):
+    """Generate rainbow colors across 0-255 positions."""
+    if pos < 85:
+        return Color(pos * 3, 255 - pos * 3, 0)
```
Comment (okalachev) [inline ws281x/examples/example.py:20]: This won't work anyway, `Color` is not defined.

File: ws281x/examples/example.py [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
 
-fill_strip(200, 200, 200)
-rospy.sleep(1)
-fill_strip(0, 200, 0)
-rospy.sleep(1)
-fill_strip(0, 0, 200)
-rospy.sleep(1)
-fill_strip(200, 0, 0)
-rospy.sleep(1)
-fill_strip(0, 0, 0)
```
Comment (okalachev) [inline ws281x/examples/example.py:19]: Please, don't change the example code as this is unrelated to the main change.

File: ws281x/srv/test.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import rospy
```
Comment (okalachev) [inline ws281x/srv/test.py:1]: What is the purpose of this file? Python script should not be in `srv` directory.

File: ws281x/vendor/rpi_ws281x/.gitignore [hunk 4]
```diff
@@ -1,13 +0,0 @@
-.sconsign.dblite
```
Comment (okalachev) [inline ws281x/vendor/rpi_ws281x/.gitignore:1]: `.gitignore` should not be removed.

File: ws281x/vendor/rpi_ws281x/python/.gitignore [hunk 5]
```diff
@@ -1,9 +0,0 @@
-*.pyc
```
Comment (okalachev) [inline ws281x/vendor/rpi_ws281x/python/.gitignore:1]: Should not be removed.

**Review by okalachev — CHANGES_REQUESTED** [review]
Hi, @siddharthcb! Thanks for your interest and contribution to the package!

But I see the following issues with it:

1. Your idea, as I understood, is to create a separate branch for channel-1 support; but your pull request is created against the `master` branch of the repo, so no `channel-1` branch would not be created.

    Anyway, I don't find this idea optimal. It would be not convenient to have two separate branches supported, for just one little feature. Why instead just not to create and document a ROS parameter for this feature, if it's really useful?

2. You replaced the entire readme with other content, removing all the main information, which is not good. The main readme should stay.
3. Irrelevant changes, like removing `.gitignore`, changing file permissions, adding examples should be put away from this PR.
4. Adding something to `/etc/ld.so.conf` seems like a hack a bit, it would be great to find a better solution.

**Comment by siddharthcb** [comment]
hello @okalachev ,
sorry for the confusion and I understand the changes to be made.
please ignore the test.py and changes made to example script since i just used it for testing and forgot to delete it from here.
I will close this pull request for now and open a new one with optimal solution for channel 1.

**Comment by okalachev** [comment]
@siddharthcb, remaking the pull request is not necessary. You can keep this PR and edit your `channel-1` branch.

**Pull request closed without merging**